### ISSUE TITLE
Easy Turtle-esque Vector API for Custom Shape Creation

### ISSUE THREAD

**Issue by philmaker1** [xeolabs/xeoengine#22]
Doesn't have to be part of the XEO API but this interests me greatly.

**Comment by xeolabs** [comment]
Yes this would be handy. Also good for procedurally-defined shapes like trees etc - could be a builder singleton with a fluent API, like the [XEO.debug](http://xeoengine.org/examples/#debugging_helper) utility.

**Comment by xeolabs** [comment]
@philmaker1 when you get a chance, could you drop in some code of what you envisage the API to look like?
